### PR TITLE
Check that env var DEVELOPER_MODE is set and 'true' before enabling developer mode

### LINKED ISSUE #673: DEVELOPER_MODE starctl option is enabled when value other than 'true' is used
When running stargate via k8 I added the following environment variable into the deployment descriptor:

* env:
  * name: DEVELOPER_MODE
    value: "false"

This still enabled developer mode when it shouldn't according to the documentation. The only way to disable developer mode was to remove the DEVELOPER_MODE environment variable from the deployment.

### PR DESCRIPTION
**Pull request by dougwettlaufer**
resolves #673

### CODE REVIEW

File: starctl [hunk 1]
```diff
@@ -132,7 +132,7 @@ if [[ ${#STARGATE_ARGS[@]} -eq 0 ]]; then
         STARGATE_ARGS+=("--proxy-port" "$PROXY_PORT")
     fi
 
-    if [[ ! -z "$DEVELOPER_MODE" ]]; then
+    if [[ ! -z "$DEVELOPER_MODE" ]] && [ "$DEVELOPER_MODE" = true ]; then
```
Comment (mpenick) [inline starctl:135]: `DEVELOPER_MODE=1` should probably work too.

Comment (dimas-b) [reply starctl:135]: Agree

Comment (dimas-b) [reply starctl:135]: I think `[[ ! -z "$DEVELOPER_MODE" ]]` is not needed now that we have  `&&` and a test for a non-empty value. 

Also the second check should probably use double brackets for consistency with other `if` statements.

Comment (dougwettlaufer) [reply starctl:135]: Double bracket and single bracket tests in bash always trip me up.

**Review by dimas-b — APPROVED** [review]

**Pull request merged**